components/suggest/src/util.rs

```diff
@@ -204,12 +205,13 @@ fn filter_map_chunks_recurse<T: Clone>(
             .iter()
             .fold(chunk_size - 1, |memo, w| memo + w.len());
         let chunk = &remaining_query[..chunk_char_len];
```

Comment:
Not related to this PR, but I found `chunk_char_len` and the comment "char length" a little bit misleading during the review because char(acter) length and byte length mean different things to the utf-8 based strings in [Rust](https://doc.rust-lang.org/nightly/book/ch08-02-strings.html#slicing-strings).

Specifically, since `w.len()` returns the byte length of `w` (v.s. the char lengh `w.chars().count()`), calling it `chunk_byte_len` is perhaps more precise than `chunk_char_len` in this case and slicing it via `&remaining_query[..chunk_byte_len]` would be perfectly sound than `&remaining_query[..query_char_len]` b/c the latter could panic if `query_char_len` is a char length.